amcl/src/amcl/pf/pf.c

```diff
@@ -512,8 +517,16 @@ int pf_resample_limit(pf_t *pf, int k)
   double a, b, c, x;
   int n;
 
+  assert(k > 0);
+
+  if (pf->limit_cache[k-1] > 0)
```

Comment:
before we access limit_cache[k-1]  we really should validate that k <= max_samples to avoid memory corruption